src/backend/commands/analyze.c

```diff
@@ -2571,7 +2571,9 @@ acquire_sample_rows_dispatcher(Relation onerel, bool inh, int elevel,
 	 * may result in different behaviour under different acl configuration.
 	 */
 	initStringInfo(&str);
-	appendStringInfo(&str, "select pg_catalog.gp_acquire_sample_rows(%u, %d, '%s');",
+	appendStringInfo(&str, "select gp_acquire_sample_rows from "
+				"(select set_config('extra_float_digits', '2', true)) sc, "
```

Comment:
@darthunix what do you think about this? Is `set_config` guaranteed to affect only the current query?

Reply:
https://www.postgresql.org/docs/9.4/functions-admin.html
> set_config sets the parameter setting_name to new_value. If is_local is true, the new value will only apply to the current transaction. If you want the new value to apply for the current session, use false instead. The function corresponds to the SQL command SET. 

But anyway I am working on getting `gp_acquire_sample_rows` result in binary mode.